### PR TITLE
Ensures that any directories specifed as part of ``outdir`` exist.

### PR DESCRIPTION
**Pull request by mabruzzo**
This is accomplished through the new ``Ensure_Outdir_Exists`` function.

Along the way, I introduced a helper function ``Is_Root_Proc()``, local to ``io.cpp`` so that we can reduce some of the ``ifdef`` statements. If you don't like this, I'm happy to remove it.

### CODE REVIEW

File: src/io/io.cpp [hunk 1]
```diff
@@ -6,9 +6,11 @@
 
 #include <algorithm>
 #include <ctime>
+#include <filesystem>
 #include <fstream>
 #include <iostream>
 #include <string>
+#include <vector>
```
Comment (alwinm) [inline src/io/io.cpp:13]: Basic question, what is <vector> providing in this PR?

File: src/io/io.cpp [hunk 2]
```diff
@@ -64,11 +75,9 @@ void Create_Log_File(struct parameters P)
 
 void Write_Message_To_Log_File(const char *message)
 {
-#ifdef MPI_CHOLLA
-  if (procID != 0) {
+  if (!!Is_Root_Proc()) {
```
Comment (bcaddy) [inline src/io/io.cpp:78]: Is the double negation intended? 

Also, several months ago there was a discussion on using keyword vs. character Boolean operators (! vs not, || vs or, etc). If I recall correctly we decided to use keyword operators.

File: src/io/io.cpp (hunk 1 above)
Comment (mabruzzo) [reply src/io/io.cpp:13]: that was an oversight. I thought I removed it. I'll fix it now

File: src/io/io.cpp (hunk 2 above)
Comment (mabruzzo) [reply src/io/io.cpp:78]: The double negative was definitely an oversight. Thanks for catching that.

Gotcha, I'll shift to using `not` and `or`

**Pull request merged**